v3/round_trip_with_retry_backoff.go

```diff
@@ -0,0 +1,86 @@
+package heroku
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/cenkalti/backoff"
+)
+
+// net/http RoundTripper interface, a.k.a. Transport
+// https://godoc.org/net/http#RoundTripper
+type RoundTripWithRetryBackoff struct {
```

Comment:
✅ good call -- was going to suggest in an earlier commit you allow some configuration to be passed through